### PR TITLE
[No ticket] Remove ember-data-factory-guy

### PR DESCRIPTION
**Pull request by fabmiz**
- Ticket: []
- Feature flag: n/a

## Purpose

Remove ember-data-factory-guy

## Summary of Changes

- Remove unused factories
- Refactor tests that used ember-data-factory-guy

## Side Effects

n/a

## QA Notes

n/a

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -202,10 +212,6 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Packages
     - upgrade to `ember-animated@0.8.1`
 
-### Removed
-- Tests
-    - `taxonomy` FactoryGuy factory
-
```
Comment (jamescdavis) [inline CHANGELOG.md]: But this *was* removed in 19.10.0: https://github.com/CenterForOpenScience/ember-osf-web/compare/19.9.0...19.10.0#diff-d597d2172d52d886f5b0785fe92e035c

Comment (fabmiz) [reply CHANGELOG.md]: Ah, right right! 🤕

Comment (fabmiz) [reply CHANGELOG.md]: Oh sorry, I force pushed to fix the tests and ended up wiping out your initial changelog fix.

Comment (jamescdavis) [reply CHANGELOG.md]: no worries, man!

**Pull request merged**